lib/util/itinerary.js

```diff
@@ -390,21 +390,55 @@ export function getLegIcon (leg, customIcons) {
   }
   let iconStr = leg.mode
   if (iconStr === 'CAR' && leg.rentedCar) {
-    iconStr = leg.rentedCarData.companies[0]
+    iconStr = leg.from.networks[0]
   } else if (iconStr === 'CAR' && leg.tncData) {
     iconStr = leg.tncData.company
   } else if (iconStr === 'BICYCLE' && leg.rentedBike) {
-    // placeholder for future vehicle rental refactor
-  } else if (iconStr === 'MICROMOBILITY' && leg.rentedVehicle && leg.rentedVehicleData) {
-    iconStr = leg.rentedVehicleData.companies[0]
+    iconStr = leg.from.networks[0]
+  } else if (iconStr === 'MICROMOBILITY' && leg.rentedVehicle) {
+    iconStr = leg.from.networks[0]
   }
 
   return getIcon(iconStr, customIcons)
 }
 
-export function getPlaceName (place) {
+export function getCompanyForNetwork (networkString, companies = []) {
+  const company = companies.find(co => co.id === networkString)
+  if (!company) {
+    console.warn(`No company found in config.yml that matches rented vehicle network: ${networkString}`, companies)
+  }
+  return company
+}
+
+export function getModeStringForCompany (company) {
+  switch (company.modes) {
+    case 'CAR_RENT':
+      return 'rented car'
+    case 'MICROMOBILITY_RENT':
+      return 'eScooter'
+    case 'BICYCLE_RENT':
+      return 'bike'
+    // If company offers more than one mode, default to `vehicle` string.
+    default:
+      return 'vehicle'
+  }
+}
+
+export function getPlaceName (place, companies) {
   // If address is provided (i.e. for carshare station, use it)
-  return place.address ? place.address.split(',')[0] : place.name
+  if (place.address) return place.address.split(',')[0]
+  if (place.networks && place.vertexType === 'VEHICLERENTAL') {
+    // For vehicle rental pick up, do not use the place name. Rather, use
+    // company name + vehicle type (e.g., SPIN eScooter). Place name is often just
+    // a UUID that has no relevance to the actual vehicle. For bikeshare, however,
+    // there are often hubs or bikes that have relevant names to the user.
+    const company = getCompanyForNetwork(place.networks[0], companies)
+    if (company) {
+      return `${company.label} ${getModeStringForCompany(company)}`
```

Comment:
Since a company can offer multiple modes, this may not be future-proof. Can't we just use the mode of the leg to get the label?